include/fuse.h

```diff
@@ -471,21 +471,26 @@ struct fuse_operations {
 	 * BIG NOTE: This is not equivalent to fsync().  It's not a
 	 * request to sync dirty data.
 	 *
-	 * Flush is called on each close() of a file descriptor.  So if a
-	 * filesystem wants to return write errors in close() and the file
-	 * has cached dirty data, this is a good place to write back data
-	 * and return any errors.  Since many applications ignore close()
-	 * errors this is not always useful.
+	 * Flush is called on each close() of a file descriptor, as opposed to
+	 * release which is called on the close of the last file descriptor for
+	 * a file.  One use of flush is to return errors from delayed writes.
+	 * However, such use is non-portable, because POSIX does not require
+	 * [close] to wait for delayed I/O to complete.  Also, many
+	 * applications ignore close() errors.  For these reason, filesystems
+	 * should not assume that errors returned by flush will ever be noticed
+	 * or even delivered.
 	 *
 	 * NOTE: The flush() method may be called more than once for each
-	 * open().	This happens if more than one file descriptor refers
-	 * to an opened file due to dup(), dup2() or fork() calls.	It is
-	 * not possible to determine if a flush is final, so each flush
-	 * should be treated equally.  Multiple write-flush sequences are
-	 * relatively rare, so this shouldn't be a problem.
-	 *
-	 * Filesystems shouldn't assume that flush will always be called
-	 * after some writes, or that if will be called at all.
+	 * open().  This happens if more than one file descriptor refers to an
+	 * opened file due to dup(), dup2() or fork() calls.  It is not
+	 * possible to determine if a flush is final, so each flush should be
+	 * treated equally.  Multiple write-flush sequences are relatively
+	 * rare, so this shouldn't be a problem.
+	 *
+	 * Filesystems shouldn't assume that flush will always be called after
+	 * some writes, or that if will be called at all.
```

Comment:
"always be called after some" doesn't make sense to me. What about simple saying "shouldn't depend on flush() being called at any particular points when data has been written - it may be called multiple times or not at all"?